packages/schema/parse/src/__tests__/validate-types.spec.ts

```diff
@@ -118,6 +118,95 @@ type Bar {
 }
 `;
 
+const circularTypes1 = `
+type A {
+  prop: B!
+}
+
+type B {
+  prop: A!
+}
+`
```

Comment:
Let's add a test for:
```
type A {
  b: B!
}

type B {
  a: A
}
```
to make sure this doesn't throw an error, as it's a valid relationship.